components/property_info.rst

```diff
@@ -0,0 +1,432 @@
+.. index::
+    single: PropertyInfo
+    single: Components; PropertyInfo
+
+The PropertyInfo Component
+==========================
+
+The PropertyInfo component provides the functionality to get information about class properties using metadata of
+popular sources.
+
+While the :doc:`PropertyAccess component</components/property_access/introduction>` allows you to read and write values
+to/from objects and arrays, the PropertyInfo component works solely with class definitions to provide information
+such as data type and visibility about properties within that class.
+
+Similar to PropertyAccess, the PropertyInfo component blurs the lines between class properties such as ``$property``
+and properties defined via accessor and mutator methods such as  ``getProperty()``, ``isProperty()``, ``setProperty()``,
+``addProperty()``, ``removeProperty()``, etc.
+
+.. versionadded:: 2.8
+    The PropertyInfo component was introduced in Symfony 2.8.
+
+.. _`components-property-information-installation`:
+
+Installation
+------------
+
+You can install the component in two different ways:
+
+* :doc:`Install it via Composer</components/using_components>` (``symfony/property-info`` on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/property-info).
+
+.. include:: /components/require_autoload.rst.inc
+
+Additional dependencies may be required for some of the
+:ref:`extractors provided with this component<components-property-info-extractors>`.
+
+.. _`components-property-information-usage`:
+
+Usage
+-----
+
+The entry point of this component is a new instance of the
+:class:`PropertyInfoExtractor<Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor>` class, providing sets of
```

Comment:
the class role can be: ``:class:`Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor` ``, no need for a label as the label matches the default label.